src/modules/ConfigLoader.js

```diff
@@ -33,8 +45,7 @@ function loadOver(newVals, targetObject) {
   var k, v, vtype;
   var next;
   if (loadDepth++ > 7) {
```

Comment:
where did this threshold value of 7 come from? what does it mean?

Reply:
The value "7" represents the maximum recursion depth allowed when copying settings into the merged config json object. This value was selected as being deep enough to indicate an invalid object was passed in or an object with a cycle was passed in. The end goal is to prevent infinite recursion. The current config object has a maximum nesting depth of 4, so I judged 7 to be adequately large as to never be encountered in a valid config object.